Fix ADCP doc

cyrilleponcelet opened the pull request:
@SverreBerg , @gavinmacaulay  please find a few fixes that request comments if need be.
Those should be merged after #52 and are linked to  #51 

The commit detail is below. I fixed a few mistakes, and what is subject to comment is that ping_t was not well defined. I moved it from ADCP group to Mean_current group where it is really used. Note that in the old ADCP file I have as example, this type ping_t is named ping_reference_type_t and is defined in /Sonar group.

The detail of the commit : 

* Mean_current : fix wrong type in valid_range definition for variables like mean_platform_roll

* Mean_current : Define mean_time as a coordinate variables, remove TBD 

* Move ping_t definition from ADCP to Mean_current (only used here) and fix its definition

* Remove useless mean_time from tableBeamGroup (probably bad copy paste)

SverreBerg commented:
See comments below.

Best regards
Sverre

From: cyrilleponcelet ***@***.***>
Sent: torsdag 16. september 2021 15:15
To: ices-publications/SONAR-netCDF4 ***@***.***>
Cc: Sverre Berg ***@***.***>; Mention ***@***.***>
Subject: [ices-publications/SONAR-netCDF4] Fix ADCP doc (#53)


This message is from an EXTERNAL SENDER - be CAUTIOUS, particularly with links and attachments.

@SverreBerg<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSverreBerg&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302023050%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ePknPMpVeIg5SsgDGFnrVx0b3nFF8yzlCWAA6VyAmGc%3D&reserved=0> , @gavinmacaulay<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgavinmacaulay&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302033001%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=9ViSdUq7cfKsRtjaDkr%2FT0Tk%2B9WeTJShfnanR5jQUzg%3D&reserved=0> please find a few fixes that request comments if need be.
Those should be merged after #52<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F52&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302042972%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=L7WGZnKV%2FpJvSKnBbGbVI8Vfuo3BTrj%2FqWZEgbi3mII%3D&reserved=0> and are linked to #51<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fissues%2F51&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302042972%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=xEhp%2F1BBFAHhpy3lgOjxXwwbTX7RGV0u6VddwIEzaKs%3D&reserved=0>

The commit detail is below. I fixed a few mistakes, and what is subject to comment is that ping_t was not well defined. I moved it from ADCP group to Mean_current group where it is really used. Note that in the old ADCP file I have as example, this type ping_t is named ping_reference_type_t and is defined in /Sonar group.

The detail of the commit :

  *   Mean_current : fix wrong type in valid_range definition for variables like mean_platform_roll
Isn't this the same valid_range definition as f.ex. platform_pitch(ping_time)?
  *   Mean_current : Define mean_time as a coordinate variables, remove TBD
  *   Move ping_t definition from ADCP to Mean_current (only used here) and fix its definition
OK
Should this be defined similar to uint64 ping_time(ping_time)
  *   Remove useless mean_time from tableBeamGroup (probably bad copy paste)
Agree

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/ices-publications/SONAR-netCDF4/pull/53<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302052918%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=q2bWLv2ElvRLVFIbhZl8iJJK0cy3lR4n43Uzxk0KaFk%3D&reserved=0>

Commit Summary

  *   Minor fixes (typos mainly)
  *   Fix missing or trailings spaces on ADCP group and subgroups
  *   Mean_current : fix wrong type in valid_range definition for variables like mean_platform_roll

File Changes

  *   M docs/tableADCP_sub_group.adoc<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53%2Ffiles%23diff-42377105b59d345b8841ebd8220db5152087a3cbde7d7da9f34dbebb57bafc66&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302052918%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tNerEyAmdCzXNhqfuckns8qKz%2F2UKajA%2B6pTr%2FsHmBg%3D&reserved=0> (84)
  *   M docs/tableAttitude_sub_group.adoc<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53%2Ffiles%23diff-6ed8434a28acbdf922cf91ae3758dc18bba4213de9090c478c9701b2c3b255fc&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302062874%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=7tQ1yBDpUHxPbRFwXOZNF%2BSiw6JaJcYauDAp6%2FNbYqs%3D&reserved=0> (2)
  *   M docs/tableBeamGridGroup1.adoc<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53%2Ffiles%23diff-116a175916a4547d934d395bbdf2fccf009975a4a60f8821617d69605b5c3219&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302062874%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=gLxsjHqVzgukjMsKlAN9BTCxacMFoLCDGExLnRqgsVc%3D&reserved=0> (16)
  *   M docs/tableBeamGroup1.adoc<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53%2Ffiles%23diff-2cea6c0cdc70c4c8a9d34cbcb8434f16cc8cb90976e7b097f22748f9bd5e2a44&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302072829%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=45OSTphfR0XgeLLIV9FscFIxc8Az7wJN%2FHFXb52420I%3D&reserved=0> (15)
  *   M docs/tableMean_current_sub_group.adoc<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53%2Ffiles%23diff-824c75005c066631437d4622b233c909b92094894fe7043116ca8826ac2c1871&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302072829%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=fZH7kpJACHIP8%2FyCcvx9UlMkxhB5lc4IIkhPuHkny4Y%3D&reserved=0> (87)

Patch Links:

  *   https://github.com/ices-publications/SONAR-netCDF4/pull/53.patch<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53.patch&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302082782%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Vf0cKoHrwVJV4VXsWhpg2%2FxYo77Y1gwcDQa1owln1eA%3D&reserved=0>
  *   https://github.com/ices-publications/SONAR-netCDF4/pull/53.diff<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53.diff&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302082782%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=gflJOcfqqMEB3ibEM46E%2BkG5KnuQlLT6H1BbpYF%2BXYc%3D&reserved=0>

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fices-publications%2FSONAR-netCDF4%2Fpull%2F53&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302092735%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=TA7oRMdjZKn7b512stAGIuVBuvEyhV9q5BB0STNQ8lg%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMLDLTFATDGV5G3UDHY2J4DUCHUWNANCNFSM5EEXSQ4Q&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302102699%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=YlnaHQk%2B%2FWtR9jxFcAZGC9mFJS%2FsMxFywHlr%2FnuKexc%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302102699%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=lutZlHZ7I5ysijgTo0g%2FYkENfKlhAqPy6ePgSfQ6K0E%3D&reserved=0> or Android<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Csverre.berg%40simrad.com%7Cd79720e8760247afb65b08d979140cb6%7Ca113bc9e1024489c902e5ac8b5fd41ce%7C0%7C0%7C637673949302112651%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Fr4xeGcqzU4oKzWiy6XZRVTn6PdNWjzbNco0iyqhR7c%3D&reserved=0>.

cyrilleponcelet commented:
*   Mean_current : fix wrong type in valid_range definition for variables like mean_platform_roll
Isn't this the same valid_range definition as f.ex. platform_pitch(ping_time)?

-> I changed both, datatype was a double but valid_range was defined as float. I changed to use the same type.

  *   Move ping_t definition from ADCP to Mean_current (only used here) and fix its definition
OK
Should this be defined similar to uint64 ping_time(ping_time)

-> It is the same type uint64 but data are used as vlen in ping_averaged. That's why need to define a new type uint64(*) to store those arrays. I think that the comment associated ("Reference to pings (by ping_time)) with ping_averaged implies clearly that units, calendar, etc. implies that the data are the same as ping_time.

Regards
